File: 3_Programação_RPi/3.5_SPI/Ex1.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <wiringPi.h>
#include <wiringPiSPI.h>
#include <unistd.h>

int spi_fd;
void ctrl_c(int sig)
{
	close(spi_fd);
	exit(-1);
}

int main(void)
{
	unsigned char user_input=1, send_microcontroller;

	signal(SIGINT, ctrl_c);
	if(wiringPiSetup() == -1)
	{
		puts("Erro em wiringPiSetup().");
		return -1;
	}
	spi_fd = wiringPiSPISetup (0, 500000);
	if(spi_fd==-1)
	{
		puts("Erro abrindo a SPI. Garanta que ela nao");
		puts("esteja sendo usada por outra aplicacao.");
		return -1;
	}
	while(user_input!=0)
	{
		puts("Digite um numero entre 1 e 5 para mandar");
		puts("o microcontrolador piscar seus LEDs,");
		puts("ou digite 0 para terminar o programa. ");
		scanf("%d", &user_input);
		if((user_input<0) || (user_input>5))
			puts("Valor invalido");
		else if(user_input>0)
		{
			send_microcontroller = user_input;
			wiringPiSPIDataRW(0, &send_microcontroller, 1);
			printf("Retorno do microcontrolador = %d\n", send_microcontroller);
			sleep(1+user_input/2);
		}
		puts("");
	}
	close(spi_fd);
	return 0;
}

File: 3_Programação_RPi/3.5_SPI/Python/Ex1.py
from spidev import SpiDev
from time import sleep

spi = SpiDev()
spi.open(0,0)
spi.max_speed_hz = 500000

user_input = 1
while user_input!=0:
	print("Digite um numero entre 1 e 5 para mandar")
	print("o microcontrolador piscar seus LEDs,")
	print("ou digite 0 para terminar o programa. ")
	user_input = int(input())
	if((user_input<0) or (user_input>5)):
		print("Valor invalido")
	else:
		if(user_input>0):
			send_microcontroller = [user_input]
			spi.xfer(send_microcontroller)
			sleep(int(1+user_input/2))
			print("Retorno do microcontrolador = %d" % send_microcontroller[0])
	print("")

spi.close()
